lib/bootloader_setup.pm

```diff
@@ -73,7 +73,11 @@ use constant GRUB_DEFAULT_FILE => "/etc/default/grub";
 # prevent grub2 timeout; 'esc' would be cleaner, but grub2-efi falls to the menu then
 # 'up' also works in textmode and UEFI menues.
 sub stop_grub_timeout {
-    send_key 'up';
+    if (assert_screen [qw(grub2 bootloader)], 600) {
+        send_key_until_needlematch('bootloader-timeout-disabled', 'up', 8, 1);
+        assert_screen 'bootloader-timeout-disabled';
```

Comment:
This second assert_screen should not be necessary.

```perl
sub send_key_until_needlematch {
    my ($tag, $key, $counter, $timeout) = @_;

    $counter //= 20;
    $timeout //= 1;
    while (!check_screen($tag, $timeout)) {
        wait_screen_change {
            send_key $key;
        };
        if (!$counter--) {
            assert_screen $tag, 1;
        }    
    }    
}
```

After all retries, it will assert_screen.

Reply:
we need to check again  because the needle match without 100% means this could be matched too earlier than timeout goooooot disabled really. That's why I spend so long to check that disabled status on grub2 by using a new created needle.

this is not normal needle match here.

Reply:
and remove needles grub2, bootloader is too dangerous at moment. I don't want break tests and this is also very hard to maintain the changes over huge number of related tests.

Reply:
https://openqa.opensuse.org/tests/1199343#step/reboot_after_installation/4 shows clear that sometimes it matches 97%, but grub2 timeout hasn't been disabled yet, for the second check then 100% which is safe.